### ISSUE TITLE
Current implementation of Toy language

### ISSUE THREAD

**Issue by jan-christiansen** [sagr4019/ResearchProject#68]
What is the accuracy when training and testing the current implementation of the Toy language?

**Comment by jan-christiansen** [comment]
When I have tested the implementation I have used the following invocations.

```
python3 preprocess_data_seq_embed.py ../DataGeneration/data/valid/ ../DataGeneration/data/not_valid/ data.npz
python3 train_keras.py data.npz
```

**Comment by Zermy** [comment]
![unbenannt](https://user-images.githubusercontent.com/16175376/42454072-dc392b36-838e-11e8-95ba-c848b832a172.PNG)

First try with python3 preprocess_data_seq_embed.py ../DataGeneration/data/valid/ ../DataGeneration/data/not_valid/ data.npz  and python3 train_seq_embed.py data.npz

**Comment by Zermy** [comment]
Train on 4000 samples, validate on 1000 samples. But i created a dataset of 20000 valid and 15000 invalid codes

**Comment by Zermy** [comment]
Data generation with 20000 valid and 15000invalid programms.

![unbenannt](https://user-images.githubusercontent.com/16175376/42458010-43dd3e08-8399-11e8-9271-d01540b1c438.PNG)

**Comment by Zermy** [comment]
Data generation with 40000 valid and 20000 invalid programms.
![unbenannt](https://user-images.githubusercontent.com/16175376/42462560-ee9bb6d0-83a3-11e8-87e4-97cc9c2a40a6.PNG)

**Comment by Zermy** [comment]
Try with
''' python preprocess_data.py /Path/to/valid/samples /Path/to/invalid/samples data.npz
python train_keras.py data.npz '''

Data generation with 40000 valid and 20000 invalid programms.
Takes about 3 hours.

**Comment by Zermy** [comment]
Last result '''python preprocess_data_string.py /Path/to/valid/samples /Path/to/invalid/samples data.npz
python train_keras.py data.npz'''

Data generation with 40000 valid and 20000 invalid programms.
![unbenannt](https://user-images.githubusercontent.com/16175376/42467520-6db8132e-83b2-11e8-9066-ef5fcdd0eda8.PNG)

accuracy is 0.6676 but i abort after a few minutes.

**Comment by jan-christiansen** [comment]
Some comments concerning the style of your comments.
* Please use code formatting for invocations of python
* The pictures are making me nuts because they are difficult to read. Why don't you simply paste the text from the terminal into the comment (in code formatting)?

**Comment by Zermy** [comment]
The code tags have been added.

**Comment by jan-christiansen** [comment]
We can just talk about the summary on the next meeting.

**Comment by jan-christiansen** [comment]
@sagr4019 Please add the steps that you will do in the course of this issue.

**Comment by sagr4019** [comment]
Tested with 10.000 valid and 10.000 invalid examples.
1. ```python data_generation.py --valid 10000 --invalid 10000 --out out --seed 42```
2. ```python preprocess_data_seq_embed.py out/valid out/invalid data.npz```
3. ```python train_seq_embed.py data.npz```
Manually aborted after the 9. epoch because of a constant categorical accuracy of 0.5000
```
Epoch 1/100
16000/16000 [==============================] - 239s 15ms/step - loss: 8.0336 - categorical_accuracy: 0.4995 - val_loss: 8.0590 - val_categorical_accuracy: 0.5000
Epoch 2/100
16000/16000 [==============================] - 281s 18ms/step - loss: 8.0590 - categorical_accuracy: 0.5000 - val_loss: 8.0590 - val_categorical_accuracy: 0.5000
Epoch 3/100
16000/16000 [==============================] - 302s 19ms/step - loss: 8.0590 - categorical_accuracy: 0.5000 - val_loss: 8.0590 - val_categorical_accuracy: 0.5000
Epoch 4/100
16000/16000 [==============================] - 292s 18ms/step - loss: 8.0590 - categorical_accuracy: 0.5000 - val_loss: 8.0590 - val_categorical_accuracy: 0.5000
Epoch 5/100
16000/16000 [==============================] - 294s 18ms/step - loss: 8.0590 - categorical_accuracy: 0.5000 - val_loss: 8.0590 - val_categorical_accuracy: 0.5000
Epoch 6/100
16000/16000 [==============================] - 294s 18ms/step - loss: 8.0590 - categorical_accuracy: 0.5000 - val_loss: 8.0590 - val_categorical_accuracy: 0.5000
Epoch 7/100
16000/16000 [==============================] - 295s 18ms/step - loss: 8.0590 - categorical_accuracy: 0.5000 - val_loss: 8.0590 - val_categorical_accuracy: 0.5000
Epoch 8/100
16000/16000 [==============================] - 284s 18ms/step - loss: 8.0590 - categorical_accuracy: 0.5000 - val_loss: 8.0590 - val_categorical_accuracy: 0.5000
Epoch 9/100
16000/16000 [==============================] - 281s 18ms/step - loss: 8.0590 - categorical_accuracy: 0.5000 - val_loss: 8.0590 - val_categorical_accuracy: 0.5000
```

**Comment by jan-christiansen** [comment]
Please also check what happens if you have more valid than invalid data set and vice versa.

**Comment by Zermy** [comment]
`320/320 [==============================] - 4s 12ms/step - loss: 8.0590 - categorical_accuracy: 0.5000 - val_loss: 8.0590 - val_categorical_accuracy: 0.5000
Epoch 10/100
320/320 [==============================] - 4s 12ms/step - loss: 8.0590 - categorical_accuracy: 0.5000 - val_loss: 8.0590 - val_categorical_accuracy: 0.5000
Epoch 11/100
320/320 [==============================] - 4s 12ms/step - loss: 8.0590 - categorical_accuracy: 0.5000 - val_loss: 8.0590 - val_categorical_accuracy: 0.5000
Epoch 12/100
320/320 [==============================] - 4s 12ms/step - loss: 8.0590 - categorical_accuracy: 0.5000 - val_loss: 8.0590 - val_categorical_accuracy: 0.5000
Epoch 13/100
320/320 [==============================] - 4s 13ms/step - loss: 8.0590 - categorical_accuracy: 0.5000 - val_loss: 8.0590 - val_categorical_accuracy: 0.5000
Epoch 14/100
320/320 [==============================] - 4s 13ms/step - loss: 8.0590 - categorical_accuracy: 0.5000 - val_loss: 8.0590 - val_categorical_accuracy: 0.5000
Epoch 15/100
320/320 [==============================] - 4s 13ms/step - loss: 8.0590 - categorical_accuracy: 0.5000 - val_loss: 8.0590 - val_categorical_accuracy: 0.5000
Epoch 16/100
320/320 [==============================] - 4s 13ms/step - loss: 8.0590 - categorical_accuracy: 0.5000 - val_loss: 8.0590 - val_categorical_accuracy: 0.5000
Epoch 17/100
320/320 [==============================] - 4s 13ms/step - loss: 8.0590 - categorical_accuracy: 0.5000 - val_loss: 8.0590 - val_categorical_accuracy: 0.5000
Epoch 18/100
320/320 [==============================] - 4s 13ms/step - loss: 8.0590 - categorical_accuracy: 0.5000 - val_loss: 8.0590 - val_categorical_accuracy: 0.5000
Epoch 19/100
320/320 [==============================] - 4s 13ms/step - loss: 8.0590 - categorical_accuracy: 0.5000 - val_loss: 8.0590 - val_categorical_accuracy: 0.5000
Epoch 20/100
320/320 [==============================] - 4s 14ms/step - loss: 8.0590 - categorical_accuracy: 0.5000 - val_loss: 8.0590 - val_categorical_accuracy: 0.5000
Epoch 21/100
320/320 [==============================] - 4s 13ms/step - loss: 8.0590 - categorical_accuracy: 0.5000 - val_loss: 8.0590 - val_categorical_accuracy: 0.5000
Epoch 22/100
320/320 [==============================] - 4s 14ms/step - loss: 8.0590 - categorical_accuracy: 0.5000 - val_loss: 8.0590 - val_categorical_accuracy: 0.5000
Epoch 23/100
320/320 [==============================] - 4s 14ms/step - loss: 8.0590 - categorical_accuracy: 0.5000 - val_loss: 8.0590 - val_categorical_accuracy: 0.5000
Epoch 24/100
320/320 [==============================] - 4s 14ms/step - loss: 8.0590 - categorical_accuracy: 0.5000 - val_loss: 8.0590 - val_categorical_accuracy: 0.5000
Epoch 25/100
320/320 [==============================] - 4s 14ms/step - loss: 8.0590 - categorical_accuracy: 0.5000 - val_loss: 8.0590 - val_categorical_accuracy: 0.5000
Epoch 26/100
320/320 [==============================] - 4s 14ms/step - loss: 8.0590 - categorical_accuracy: 0.5000 - val_loss: 8.0590 - val_categorical_accuracy: 0.5000
Epoch 27/100`
